web/src/actions/activities.js

```diff
@@ -21,7 +22,20 @@ export const UPDATE_ACTIVITY = 'UPDATE_ACTIVITY';
 const actionWithYears = (type, other) => (dispatch, getState) =>
   dispatch({ type, ...other, years: getState().apd.data.years });
 
-export const addActivity = () => actionWithYears(ADD_ACTIVITY);
+export const addActivity = () => (dispatch, getState) => {
+  const oldActivities = getState().activities.allKeys;
```

Comment:
slightly more efficient way to do so this to avoid the O(n*n) filter (and take advantage of `find` method):

```
const oldKeys = new Set(getState().activities.allKeys);

...

const newKey = getState().activities.allKeys.find(k => !oldKeys.has(k))

...
```

Reply:
One question about using a `Set` - do you know if it'd cause us problems with IE support?  I get that it's supported in babel-polyfill but I'm not sure how we're pulling that in, so...  😄 

In any case, `find` is definitely better than `filter`.  👍